pkg/ssh/client/ssh.go

```diff
@@ -133,8 +133,12 @@ func importKey(ctx context.Context, keyName, fingerprint string, keyContent *str
 	if existing, err := findKeyInEC2(ctx, ec2API, keyName); err != nil {
 		return err
 	} else if existing != nil {
-		if *existing.KeyFingerprint != fingerprint {
-			return fmt.Errorf("SSH public key %s already exists, but fingerprints don't match (exected: %q, got: %q)", keyName, fingerprint, *existing.KeyFingerprint)
+		// AWS uses a slightly different encoding than what Go is doing and ends up with an extra padding.
+		// They are using a combination of base64 and sha256 together. Since sha256 generates 32 bytes,
+		// base64 pads it to 33 with an extra `=`. The padding is safe to remove.
```

Comment:
Both AWS and eksctl use the same algorithm to generate the fingerprint for Ed25519 keys, but the Go implementation uses an unpadded base64 encoding (`base64.RawStdEncoding`), which is why the eksctl-generated fingerprint has no padding (=).